lte/gateway/c/core/oai/test/sgw_s8_task/sgw_csr.cpp

```diff
@@ -137,53 +127,180 @@ TEST_F(SgwS8Config, update_pdn_session_on_cs_rsp) {
   fill_itti_csreq(&session_req, default_eps_bearer_id);
 
-  sgw_s11_tunnel.remote_teid = session_req.sender_fteid_for_cp.teid;
   sgw_update_bearer_context_information_on_csreq(
-      sgw_state, sgw_pdn_session, sgw_s11_tunnel, &session_req, imsi64);
+      sgw_state, sgw_pdn_session, &session_req, imsi64);
 
   s8_create_session_response_t csresp = {0};
-  fill_itti_csrsp(&csresp, sgw_s11_tunnel.local_teid);
+  fill_itti_csrsp(&csresp, temporary_session_id);
   csresp.eps_bearer_id = 7;  // Send wrong eps_bearer_id
   EXPECT_EQ(
       sgw_update_bearer_context_information_on_csrsp(sgw_pdn_session, &csresp),
       RETURNerror);
   sgw_state_exit();
+  std::this_thread::sleep_for(
+      std::chrono::milliseconds(END_OF_TESTCASE_SLEEP_MS));
+}
+
+// TC validates deletion of pdn session
+TEST_F(SgwS8ConfigAndCreateMock, delete_session_req_handling) {
+  sgw_state_t* sgw_state = get_sgw_state(false);
+
+  sgw_eps_bearer_context_information_t* sgw_pdn_session = NULL;
+  uint32_t temporary_session_id                         = 0;
+  sgw_pdn_session = sgw_create_bearer_context_information_in_collection(
+      sgw_state, &temporary_session_id);
+
+  itti_s11_create_session_request_t session_req = {0};
+  fill_itti_csreq(&session_req, default_eps_bearer_id);
+
+  sgw_update_bearer_context_information_on_csreq(
+      sgw_state, sgw_pdn_session, &session_req, imsi64);
+
+  EXPECT_EQ(strcmp(sgw_pdn_session->pdn_connection.apn_in_use, "NO APN"), 0);
+
+  s8_create_session_response_t csresp = {0};
+  fill_itti_csrsp(&csresp, temporary_session_id);
+
+  EXPECT_CALL(*mme_app_handler, mme_app_handle_create_sess_resp()).Times(1);
+  EXPECT_EQ(
+      sgw_s8_handle_create_session_response(sgw_state, &csresp, imsi64),
+      RETURNok);
+
+  EXPECT_TRUE((sgw_get_sgw_eps_bearer_context(csresp.context_teid)) != nullptr);
+  sgw_eps_bearer_ctxt_t* bearer_ctx_p = sgw_cm_get_eps_bearer_entry(
+      &sgw_pdn_session->pdn_connection, csresp.eps_bearer_id);
+  EXPECT_TRUE(bearer_ctx_p != nullptr);
+
+  EXPECT_TRUE(
+      bearer_ctx_p->paa.ipv4_address.s_addr == csresp.paa.ipv4_address.s_addr);
+  EXPECT_TRUE(
+      bearer_ctx_p->p_gw_teid_S5_S8_up ==
+      csresp.bearer_context[0].pgw_s8_up.teid);
+  itti_s11_delete_session_request_t ds_req = {0};
+  fill_delete_session_request(
+      &ds_req, csresp.context_teid, csresp.eps_bearer_id);
+
+  EXPECT_EQ(
+      sgw_s8_handle_s11_delete_session_request(sgw_state, &ds_req, imsi64),
+      RETURNok);
+
+  s8_delete_session_response_t ds_rsp = {0};
+  fill_delete_session_response(&ds_rsp, csresp.context_teid, REQUEST_ACCEPTED);
+  EXPECT_CALL(*mme_app_handler, mme_app_handle_delete_sess_rsp()).Times(1);
+  EXPECT_EQ(
+      sgw_s8_handle_delete_session_response(sgw_state, &ds_rsp, imsi64),
+      RETURNok);
+  EXPECT_TRUE((sgw_get_sgw_eps_bearer_context(csresp.context_teid)) == nullptr);
+  sgw_state_exit();
+  std::this_thread::sleep_for(
+      std::chrono::milliseconds(END_OF_TESTCASE_SLEEP_MS));
+}
```

Comment:
can you add some comments in different blocks of the tests to easily understand what is going on.

In the future, when you will have to modify the test and you have forgotten about it,  having those comments will be very very useful.